### ISSUE TITLE
Bug: no-unsupported-features/es-builtins false positives

### ISSUE THREAD

**Issue by aladdin-add** [eslint-community/eslint-plugin-n#260]
### Environment

Node version:v22.0.0
npm version:v10.4.0
ESLint version:v9.0.0
eslint-plugin-n version: 17.3.1
Operating System:darwin 23.5.0

### What rule do you want to report?

no-unsupported-features/es-builtins

### Link to Minimal Reproducible Example

n/w

### What did you expect to happen?

```js
'use strict';
fetch('/api/xxx');
```

result:
```bash
  2:1  error  The 'fetch' is still an experimental feature and is not supported until Node.js 18.0.0. The configured version range is '>=17'  n/no-unsupported-features/es-builtins
```

Since `fetch` is not an es builtin, the rule should not report this case.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by scagood** [comment]
I am unsure I agree, as I thought fetch is both es builtin and a nodew built in 🤔
